documentation/query.adoc

```diff
@@ -57,111 +57,173 @@ For example, try click on the `Product` node label and you'll see this query pop
+In cases where the same supplier supplies products from more than one category, the distinct category names are collected into a list, as you can see in the _categories_ column.
+
+Even though you specified the _Product_ node in the `MATCH` part of the query, no information about the actual product is returned since this is not included in the `RETURN` part.
 
 Maybe you are interested only in suppliers of a certain category, say _Condiments_:
 
 [source, cypher]
 ----
-MATCH (c:Category {categoryName:"Condiments"})<--(:Product)<--(s:Supplier)
-RETURN DISTINCT s.companyName as CondimentsSuppliers
+MATCH (s:Supplier)-->(:Product)-->(c:Category)
+WHERE c.categoryName = 'Condiments'
+RETURN DISTINCT s.companyName as condimentsSuppliers
 ----
 
-This query finds a category with the name _Condiments_ (one of the properties on the _Category_ node, if you recall the data model) and links that to a _Product_ and the product to a _Supplier_ that supplies it.
-The query uses _variables_, `c` and `s`, for the category and the supplier, respectively, but _not_ for the product.
-That's because the product happens to be on the path between the two types of nodes you are interested in, just like in the previous example.
-Note that since you are only interested in that a relationship exists between these nodes and not in _what type_ of relationship, you don't have to specify any relationship type in the query.
+Here we're not interested in the `Product` nodes, nor the exact types of the relationships between the nodes, so we don't have to specify them.
 
 This is fairly basic, so let's move on to something a little more complex.
 
 == Write more advanced Cypher for problem-solving
 
-With the basics out of the way, let's have a look a more complex problem.
-Assume that you want to see which product categories are typically co-ordered with other product categories.
-This might help you understand which products to promote alongside others, or finding gaps in opportunities to promote products often ordered together.
+Assume that you want to see which product categories are typically co-ordered with other product categories and how frequently.
+
+This might help you understand which products to promote alongside others.
 
 [source, cypher]
 ----
+// which categories are the products of an order in
 MATCH (o:Order)-[:ORDERS]->(:Product)-[:PART_OF]->(c:Category)
-WITH o, collect(c.categoryName) as Categories WHERE size(Categories) > 1
-RETURN DISTINCT Categories
+// retain same ordering of categories
+WITH o, c ORDER BY c.categoryName
+// aggregate categories by order into a list of names
+WITH o, collect(DISTINCT c.categoryName) as categories 
+// only orders with more than one category
+WHERE size(categories) > 1
+// count how frequently the pairings occurr
+RETURN categories, count(*) as freq
+// order by frequency
+ORDER BY freq DESC
+LIMIT 50
 ----
 
+Another question would be, which customers are similar, i.e. ordering similar products most frequently.
```

Comment:
```suggestion
Another related question is which customers are similar, i.e. ordering similar products at the same frequency.
```

Reply:
That would change the meaning.

Reply:
Hmm.. How about this:
_You can extend this and find customers who display similar purchasing patterns, i.e. which customers order similar products most frequently._